script.py

```diff
@@ -144,13 +155,16 @@ def update_validator_database_and_reports():
                 continue
         process_given_area(cursor, entry)
         connection.commit()
-        #generate_webpage_with_error_output.write_index_and_merged_entries(cursor) # update after each run
+        # generate_webpage_with_error_output.write_index_and_merged_entries(cursor) # update after each run
+    commit_and_publish_changes_in_report_directory(
+        cursor)  # note that it is called not only here! But also at start of the function
     connection.close()
-    commit_and_publish_changes_in_report_directory() # note that it is called not only here! But also at start of the function
+
```

Comment:
I moved this connection.close() to after the commit_and_publish which caused an exception as well